sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_analysis_client.py

```diff
@@ -105,18 +105,28 @@ def begin_analyze_document(
                 :caption: Analyze a custom document. For more samples see the `samples` folder.
         """
 
-        if not model_id:
```

Comment:
I might be missing something but consider this:

```
continuation_token = kwargs.pop("continuation_token", None)
if not model_id and continuation_token is None:
    raise ValueError("model_id cannot be None or empty.")
```

then we shouldn't need two separate calls to the generated method / need manage their arguments differently

Reply:
I was iffy on whether to make the if more complex for each of the value checks or just move the continuation token request up. I dont mind doing what you propose for now.

Reply:
Up to you, whatever you think is more manageable over time

Reply:
In my mind, this is simpler for the future if more checks need to be added or more complex/custom code for the real request vs the cont token request. I'll leave it like this for now then  :)